File: lcof/面试题55 - I. 二叉树的深度/Solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func maxDepth(root *TreeNode) int {
    if (root == nil) {
        return 0
    }
    left, right := maxDepth(root.Left), maxDepth(root.Right)
    if left > right {
        return 1 + left
    }
    return 1 + right
}

File: lcof/面试题55 - I. 二叉树的深度/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root is None:
            return 0
        return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))
